### PR TITLE
Adding a new function that runs the setup of structure S

### PR DESCRIPTION
**Pull request by swilliamson7**
If you're not opposed to this I added a function that sets up and returns `S` but doesn't do any integrating, I use it to start my Enzyme runs

### CODE REVIEW

**Comment by milankl** [comment]
Sounds great, yeah I mean this is where the `run_model` interface is falling apart because as you clearly say one does need a way to set all parameters for a model configuration, then have access to *the* (i.e. one single) model/simulation object and then a way to run that simulation forward. That's why we do that in SpeedyWeather. Glad you're suggesting the same!

**Comment by milankl** [comment]
What about calling `run_setup`, `model_setup` as it creates a `ModelSetup` and returns it?

**Comment by swilliamson7** [comment]
Sounds great, easy to remember! I'll merge it after the checks complete

**Review by milankl — APPROVED** [review]

**Review by milankl — APPROVED** [review]

File: src/run_model.jl [hunk 1]
```diff
@@ -39,4 +39,35 @@ function run_model(::Type{T},P::Parameter) where {T<:AbstractFloat}
 
     return Prog
 
+end
+
+function model_setup(::Type{T}=Float32;     # number format
+    kwargs...                             # all additional parameters
+    ) where {T<:AbstractFloat}
+
+    P = ShallowWaters.Parameter(T=T;kwargs...)
```
Comment (milankl) [inline src/run_model.jl:48]: Maybe you want to take out the `ShallowWaters.` declarations as they are only necessary when you defined this outside of the `ShallowWaters` scope

File: src/run_model.jl [hunk 2]
```diff
@@ -39,4 +39,35 @@ function run_model(::Type{T},P::Parameter) where {T<:AbstractFloat}
...
+
+    G = ShallowWaters.Grid{T,Tprog}(P)
+    C = ShallowWaters.Constants{T,Tprog}(P,G)
+    F = ShallowWaters.Forcing{T}(P,G)
+
+    Prog = ShallowWaters.initial_conditions(Tprog,G,P,C)
+    Diag = ShallowWaters.preallocate(T,Tprog,G)
+
+    # one structure with everything inside 
+    S = ShallowWaters.ModelSetup{T,Tprog}(P,G,C,F,Prog,Diag,0)
```
[the remark this answers is not in the document]
Comment (swilliamson7) [reply src/run_model.jl:69]: Ah whoops, I missed that

**Pull request merged**